k6ext/k6test/vu.go

```diff
@@ -22,14 +23,19 @@ import (
 // VU is a k6 VU instance.
 type VU struct {
 	*k6modulestest.VU
-	Loop *k6eventloop.EventLoop
+	Loop      *k6eventloop.EventLoop
+	runLoopMu sync.Mutex
 }
 
 // ToGojaValue is a convenience method for converting any value to a goja value.
 func (v *VU) ToGojaValue(i interface{}) goja.Value { return v.Runtime().ToValue(i) }
 
 // RunLoop is a convenience method for running fn in the event loop.
 func (v *VU) RunLoop(fn func() error) error {
+	v.runLoopMu.Lock()
+	defer v.runLoopMu.Unlock()
+
+	v.Loop.WaitOnRegistered()
```

Comment:
We're only supposed to have one eventloop per vu, which means we're not supposed to share the eventloop between threads. So each test that requires an eventloop needs its own browser/vu. So this is a bit of hack to allow multiple tests use one browser/vu and one eventloop. How do you feel about that @imiric? Would you rather we followed the eventloop design and had one browser process per test that needs an eventloop?

Reply:
I think we generally follow that rule, as otherwise we would run into issues with Goja, which also isn't thread-safe.

We were missing the `WaitOnRegistered()` call, so that's fine, but what happens if you remove the lock? If any tests fail we should rewrite them to not share the browser / eventloop, so it might be better to remove the lock and force us to do that.

Reply:
Thanks! Resolved in: 8b231216c784683f90f728dfb2bfa6ccd855d08b